scipy/sparse/_base.py

```diff
@@ -604,7 +604,7 @@ def _rmul_dispatch(self, other):
                 tr = other.transpose()
             except AttributeError:
                 tr = np.asarray(other).transpose()
-            return (self.transpose() @ tr).transpose()
+            return self.transpose()._mul_dispatch(tr).transpose()
```

Comment:
I was curious whether this is to prevent a recursive loop, but I don't think that can happen since this goes to `__mul__`. But I'm fine adding it as a precaution.

Reply:
I propose using `dot` here because I'm not sure if `self.transpose()` must preserve `_is_array` flag especially for user-defined sparse classes.  I'll try to fix #15258, too.
```suggestion
            return self.transpose().dot(tr).transpose()
```

Reply:
Sorry.  `_mul_dispatch` is better because it can return `NotImplemented`.  Now I see `_mul_dispatch` is always "matmul" (or `_mul_scalar`).